game.cpp

```diff
@@ -0,0 +1,84 @@
+#include <iostream>
+#include "randomEngine.cpp"
+#include "player.cpp"
+
+class NumberGame{
+    int randNumber;
+    RandomEngine randomEngine;
+    Player human;
+    Player pc;
+public:
+    NumberGame() : human(), pc(), randomEngine(1, 20){}
+    void run(){
+        while(human.is_playable() && pc.is_playable()) {
+            std::cout << "Now u have: " << human.getScore() << std::endl
+                      << "and pc have: " << pc.getScore() << std::endl;
+            human.addNumbers(getPlayerInput());
+            pc.addNumbers(calcComputerTurn());
+            randNumber = randomEngine.getRandomNumber();
+
+            int absComputer = std::abs(sum(pc) - randNumber);
+            int absHuman = std::abs(sum(human) - randNumber);
+
+            std::cout << "Game begin" << std::endl
+                      << "Pc chose: " ; pc.showNumbers();
+            std::cout <<  "Your choice: "; human.showNumbers();
+            std::cout << "Number: " << randNumber << std::endl;
+
+            if (absComputer == absHuman) {
+                std::cout << "*********Tie!*********" << std::endl;
+            } else {
+                defineWinnerByAbs(absComputer, absHuman);
+            }
+        }
+        std::cout << "Gj";
+    }
+
+    std::set<int> getPlayerInput(){
+        int number;
+        std::set<int> input;
+        while (input.size() < 2) {
+            std::cout << "Please enter number between 1 - 20 range: ";
+            std::cin >> number;
+            if(std::cin.fail() || number < 1 || number > 20){
+                std::cin.clear();
+                std::cin.ignore(std::numeric_limits<std::streamsize>::max(), '\n');
+                std::cout << "Invalid input. Please enter a number between 1 and 20.\n";
+            } else {
+                input.insert(number);
+            }
+        }
+        return input;
+    }
+
+    std::set<int> calcComputerTurn(){
+        int pc_value;
+        std::set<int> pc_input;
+        pc_value = randomEngine.getRandomNumber();
+        for(int i = 0; i < 2; ++i) {
```

Comment:
Can you update this part of code the same as input by human, where you use size of set for undestand about progress of inputing